satellite/overlay/piececount_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package overlay_test
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testrand"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+	"storj.io/storj/satellite/overlay"
+	"storj.io/storj/satellite/satellitedb/satellitedbtest"
+)
+
+func TestDB_PieceCounts(t *testing.T) {
+	satellitedbtest.Run(t, func(t *testing.T, db satellite.DB) {
+		ctx := testcontext.New(t)
+		defer ctx.Cleanup()
+
+		overlaydb := db.OverlayCache()
+
+		type TestNode struct {
+			ID         storj.NodeID
+			PieceCount int // TODO: fix to int64
+		}
+
+		nodes := make([]TestNode, 100)
+		for i := range nodes {
+			nodes[i].ID = testrand.NodeID()
+			nodes[i].PieceCount = int(math.Pow10(i + 1))
+		}
+
+		for _, node := range nodes {
+			require.NoError(t, overlaydb.UpdateAddress(ctx, &pb.Node{
+				Id: node.ID,
+				Address: &pb.NodeAddress{
+					Transport: pb.NodeTransport_TCP_TLS_GRPC,
+					Address:   "0.0.0.0",
+				},
+				LastIp: "0.0.0.0",
+			}, overlay.NodeSelectionConfig{}))
+		}
+
+		// check that they are initialized to zero
+		initialCounts, err := overlaydb.AllPieceCounts(ctx)
+		require.NoError(t, err)
+		require.Empty(t, initialCounts)
+		// TODO: make it actually return everything
```

Comment:
Is it fine to leave the TODO here and remove the code below?
Or better fix it directly? 👍

Reply:
I didn't want to fix it directly at this moment, otherwise the PR would grow too large. But I certainly can remove it.